libs/perun/utils/src/lib/table-wrapper/table-wrapper.component.ts

```diff
@@ -62,4 +62,33 @@ export class TableWrapperComponent implements OnInit {
       }); // scroll to top of table on page changes
     }
   }
+
+  changePage(event: FocusEvent): void {
+    const input = event.target as HTMLInputElement;
+    let selectedPage = parseInt(input.value);
+
+    if (!selectedPage || selectedPage < 1) {
+      selectedPage = 1;
+    }
+    if (selectedPage > this.paginator.getNumberOfPages()) {
+      selectedPage = this.paginator.getNumberOfPages();
+    }
+
+    this.paginator.pageIndex = selectedPage - 1;
+    const changePageEvent: PageEvent = {
+      length: this.paginator.length,
+      pageSize: this.paginator.pageSize,
+      pageIndex: this.paginator.pageIndex,
+    };
+    this.paginator.page.next(changePageEvent);
+
+    input.value = selectedPage.toString();
+  }
+
+  onlyValidKeys(event: KeyboardEvent): boolean {
+    const charCode = event.key.charCodeAt(0);
+    return (
+      charCode === 65 || charCode === 66 || charCode === 68 || (charCode >= 48 && charCode <= 57)
```

Comment:
Either use literals or describe what this does in a comment

Reply:
Also why are A B and D allowed? I guess for some shortcuts? In that case allow them in combination with `CTRL`? This whole check seems unnecessary, IMO resetting the input if invalid value was put in would be enough, but up to you.